### PR TITLE
[ACR] Add support for VNet/IP rules

### PR DESCRIPTION
**Pull request by djyou**
Fixes https://github.com/AzureCR/azure-cli/issues/297

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/network_rule.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.util import CLIError
+
+from azure.mgmt.containerregistry.v2018_09_01.models import (
```
Comment (tjprescott) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/network_rule.py]: You cannot import this with a hard coded version. You must update the module to use a multi-API-versioned SDK.

Comment (djyou) [reply src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/network_rule.py]: @tjprescott We had offline discussions with @yugangw-msft We will clean this up plus remove deprecated legacy commands as a separate PR, tracked by this issue https://github.com/Azure/azure-cli/issues/8265

Comment (tjprescott) [reply src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/network_rule.py]: Sounds good @djyou.  AKS team is following the same approach.

**Pull request merged**